### PR TITLE
Ignore crab king in dev mode

### PR DESCRIPTION
**Pull request by haythem96**

### CODE REVIEW

File: pages/api/auction/createOrEditAuction.ts [hunk 1]
```diff
@@ -8,11 +8,13 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   const { signature, auction } = req.body
   const owner = await crabV2Contract.owner()
 
-  try {
-    const isOwner = verifyMessage(KING_CRAB, signature, owner)
-    if (!isOwner) return res.status(401).json({ message: 'Not owner' })
-  } catch (e) {
-    return res.status(401).json({ message: "Signature can't be verified" })
+  if (process.env.environment != 'dev') {
```
Comment (KMKoushik) [inline pages/api/auction/createOrEditAuction.ts:11]: let's use process.env.NODE_ENV instead. It's already provided by next

**Review by KMKoushik — CHANGES_REQUESTED** [review]

File: pages/api/auction/createOrEditAuction.ts (hunk 1 above)
Comment (haythem96) [reply pages/api/auction/createOrEditAuction.ts]: Done

File: pages/api/auction/createOrEditBid.ts [hunk 2]
```diff
@@ -11,13 +11,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+    const oldBid = auction.bids[`${order.trader}-${order.nonce}`] as Bid
+
+    try {
+      const isOwner = verifyOrder(order, signature, oldBid?.bidder || order.trader)
+      if (!isOwner) return res.status(401).json({ message: 'Not owner' })
+    } catch (e) {
+      return res.status(401).json({ message: "Signature can't be verified" })
+    }
+  } else {
+    auction.bids = {}
```
Comment (KMKoushik) [inline pages/api/auction/createOrEditBid.ts:25]: I don't understand this case. Is there a situation where auction.bids is undefined?

Comment (haythem96) [reply pages/api/auction/createOrEditBid.ts:25]: I was testing the submit_bid in the SDK, so I created quickly a new auction https://github.com/opynfinance/crab-v2-sdk-python/blob/a868e353bed834830a55d2bdf26833f9a53f81eb/scripts/crab_v2_sdk_script.py#L68 without passing the bids array, I think for now we can have this if/else statement to make it easier to test, until more improvement to the SDK tests.

Comment (haythem96) [reply pages/api/auction/createOrEditBid.ts:25]: Also the API also should later have input validation checks, an auction without bids array shouldn't be created IMO if we are not counting a special case for it

**Review by KMKoushik — APPROVED** [review]

**Pull request merged**